kamelet-support/src/test/java/io/kaoto/backend/metadata/parser/step/camelRoute/CamelRouteParseCatalogTest.java

```diff
@@ -0,0 +1,176 @@
+import io.kaoto.backend.model.parameter.StringParameter;
+import io.kaoto.backend.model.step.Step;
+import io.quarkus.test.junit.QuarkusTest;
+import org.junit.jupiter.api.Disabled;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.function.BiFunction;
+import java.util.stream.Collectors;
+
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertIterableEquals;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@QuarkusTest
+public class CamelRouteParseCatalogTest {
+
+    @Inject
+    public void setParseCatalog(final CamelRouteParseCatalog parseCatalog) {
+        this.parseCatalog = parseCatalog;
+    }
+    private CamelRouteParseCatalog parseCatalog;
+
+    @Disabled("Git is down")
+    @Test
+    void shouldLoadFromGit() {
+        String camelGit = "https://github.com/apache/camel.git";
+        String tag = "camel-3.18.2";
+        InMemoryCatalog<Step> catalog = new InMemoryCatalog<>();
+
+        ParseCatalog<Step> camelParser =
+                parseCatalog.getParser(camelGit, tag);
+        List<Step> steps = camelParser.parse().join();
+
+        assertTrue(catalog.store(steps));
+
+        Step browseComponentSink = catalog.searchStepByID("browse-sink");
+        Step browseComponentSource = catalog.searchStepByID("browse-source");
+        Step browseComponentAction = catalog.searchStepByID("browse-action");
+
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentSource, "source", false);
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentSink, "sink", false);
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentAction, "action", false);
+    }
+
+    @Test
+    void shouldLoadFromJar() {
+        String camelGit = "https://github.com/apache/camel/"
+                + "archive/refs/tags/camel-3.18.2.zip";
+        InMemoryCatalog<Step> catalog = new InMemoryCatalog<>();
+
+        ParseCatalog<Step> camelParser =
+                parseCatalog.getParser(camelGit);
+        List<Step> steps = camelParser.parse().join();
+
+        assertTrue(catalog.store(steps));
+
+        Step browseComponentSink = catalog.searchStepByID("browse-sink");
+        Step browseComponentSource = catalog.searchStepByID("browse-source");
+        Step browseComponentAction = catalog.searchStepByID("browse-action");
+
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentSource, "source", false);
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentSink, "sink", false);
+        assertBrowseJsonHasBeenParsedCorrectly(
+                browseComponentAction, "action", false);
+    }
+
+    @Test
+    void shouldLoadFromLocalFolder() {
+        String resourcesDirectory = "/home/joshiraez/dev/redhat/"
```

Comment:
Change to proper path

Reply:
This is already changed and I'm not sure why git is not loading the changes correctly?